tests/lib/rules/multiline-comment-style.js

```diff
@@ -482,6 +712,348 @@ ruleTester.run("multiline-comment-style", rule, {
+                // }
+            `,
+            options: ["separate-lines"],
+            errors: [
+                { messageId: "expectedLines", line: 2 }
+            ]
+        },
+        {
+            code: `
+                /*
+                 *{
+                 *    "foo": 1,
+                 *    "bar": 2
+                 *}
+                 */
+            `,
+            output: `
+                // {
+                //     "foo": 1,
+                //     "bar": 2
+                // }
+            `,
+            options: ["separate-lines"],
+            errors: [
+                { messageId: "expectedLines", line: 2 }
+            ]
+        },
+        {
+            code: `
+                /*
+                 *   {
+                 *       "foo": 1,
+                 *       "bar": 2
+                 *   }
+                 */
+            `,
+            output: `
+                //   {
+                //       "foo": 1,
+                //       "bar": 2
+                //   }
+            `,
+            options: ["separate-lines"],
+            errors: [
+                { messageId: "expectedLines", line: 2 }
+            ]
+        },
+        {
+            code: `
+                /*
+            *{
+                 *    "foo": 1,
+                    *    "bar": 2
+                 *}
+                  */
+            `,
+            output: `
+                // {
+                //     "foo": 1,
+                //     "bar": 2
+                // }
+            `,
+            options: ["separate-lines"],
+            errors: [
+                { messageId: "expectedLines", line: 2 }
+            ]
+        },
+        {
+            code: `
+                /*
+                 *   {
+                 *       "foo": 1,
+                 *       "bar": 2
+           *}
+                 */
+            `,
+            output: `
+                //    {
+                //        "foo": 1,
+                //        "bar": 2
+                // }
+            `,
+            options: ["separate-lines"],
+            errors: [
+                { messageId: "expectedLines", line: 2 }
+            ]
+        },
+        {
+            code: `
+                /*
+                {
+                    "foo": 1,
+                    "bar": 2
+                }
+                */
+            `,
+            output: `
+                // {
+                //     "foo": 1,
+                //     "bar": 2
+                // }
+            `,
+            options: ["separate-lines"],
+            errors: [
+                { messageId: "expectedLines", line: 2 }
+            ]
+        },
+        {
+            code: `
+                /* {
+                       "foo": 1,
+                       "bar": 2
+                   } */
+            `,
+            output: `
+                // {
+                //     "foo": 1,
+                //     "bar": 2
+                // }${" "}
```

Comment:
We could update the autofixer to not include trailing whitespace, but I wasn't sure it was worth the extra complexity. Thoughts?

I added this because many people configure their editors to remove trailing whitespace automatically, and that could lead to a frustratingly vague error when running the tests.